### PR TITLE
No delay when fetching blobs with known block and no attempt to recover blobs for unknown block

### LINKED ISSUE #8796: Remove delay when looking up blobs via local EL
Once there is enough adoption of `engine_getBlobs` we can remove the delay and do the local lookup of the missing blobs immediately once we receive the block.
RPC delay should remain unchanged.

### PR DESCRIPTION
**Pull request by StefanBratanov**
## PR Description

- Separate between RPC block fetching and RPC blobs fetching

Changed the flow to be as such:

- If we first receive a blob sidecar: Only trigger block fetch after a delay (same as before)
- If we first receive block: No delay and trigger EL lookup. RPC lookup is attempted after a delay (same as before)
- On arrival of block, if tracker exists: We trigger the blobs lookup from EL again and RPC lookup (if tracker is not complete)

## Fixed Issue(s)
fixes #8796 

## Documentation

- [X] I thought about documentation and added the `doc-change-required` label to this PR if updates are required.

## Changelog

- [X] I thought about adding a changelog entry, and added one if I deemed necessary.

### CODE REVIEW

**Comment by tbenr** [comment]
nice! I'll play around with this tomorrow

File: ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/blobs/BlockBlobSidecarsTracker.java [hunk 1]
```diff
@@ -315,22 +308,31 @@ private void printDebugTimings(final Map<UInt64, Long> debugTimings) {
...
     } else {
-      timingsReport.append("Local EL fetch wasn't required - ");
+      timingsReport.append("Local EL blobs fetch wasn't required - ");
+    }
+
+    if (debugTimings.containsKey(RPC_BLOCK_FETCH_TIMING_IDX)) {
+      timingsReport
+          .append("RPC block fetch delay ")
+          .append(debugTimings.get(RPC_BLOCK_FETCH_TIMING_IDX) - creationTime)
+          .append("ms");
```
Comment (tbenr) [inline ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/blobs/BlockBlobSidecarsTracker.java]:
```
          .append("ms - ");
    } else {
      timingsReport.append("RPC block fetch wasn't required - ");
```

**Review by tbenr — COMMENTED** [review]
As discussed I'd add an delay before requesting blobs via RPC when local el doesn't have the missing blob(s)

File: ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/blobs/BlockBlobSidecarsTracker.java [hunk 2]
```diff
@@ -315,22 +308,31 @@ private void printDebugTimings(final Map<UInt64, Long> debugTimings) {
...
+    if (debugTimings.containsKey(RPC_BLOCK_FETCH_TIMING_IDX)) {
       timingsReport
-          .append("Local EL fetch delay ")
-          .append(debugTimings.get(LOCAL_EL_FETCH_TIMING_IDX) - creationTime)
+          .append("RPC block fetch delay ")
+          .append(debugTimings.get(RPC_BLOCK_FETCH_TIMING_IDX) - creationTime)
           .append("ms - ");
     } else {
-      timingsReport.append("Local EL fetch wasn't required - ");
+      timingsReport.append("RPC block fetch wasn't required");
```
Comment (tbenr) [inline ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/blobs/BlockBlobSidecarsTracker.java]: final ` - ` here :-)

File: ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/blobs/BlockBlobSidecarsTracker.java [hunk 3]
```diff
@@ -111,29 +111,11 @@ public Optional<BlobSidecar> getBlobSidecar(final UInt64 index) {
...
-
-    // We may return maxBlobsPerBlock because we don't know the block
-    return UInt64.range(UInt64.ZERO, maxBlobsPerBlock)
-        .filter(blobIndex -> !blobSidecars.containsKey(blobIndex))
-        .map(blobIndex -> new BlobIdentifier(slotAndBlockRoot.getBlockRoot(), blobIndex));
-  }
-
-  public Stream<BlobIdentifier> getUnusedBlobSidecarsForBlock() {
     final Optional<Integer> blockCommitmentsCount = getBlockKzgCommitmentsCount();
     checkState(blockCommitmentsCount.isPresent(), "Block must me known to call this method");
```
Comment (mehdi-aouadi) [inline ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/blobs/BlockBlobSidecarsTracker.java]: Since we're at it: must **be** know

**Review by mehdi-aouadi — APPROVED** [review]
LGTM
I'd suggest running a local interop or deploy it to a Holesky node and see how it goes

**Pull request merged**